header.go

```diff
@@ -1880,8 +1894,8 @@ func (h *ResponseHeader) parseFirstLine(buf []byte) (int, error) {
 		}
 		return 0, fmt.Errorf("unexpected char at the end of status code. Response %q", buf)
 	}
-	if len(b) > n+1 && !bytes.Equal(b[n+1:], statusLine(h.statusCode)) {
-		h.SetStatusLine(b[n+1:])
+	if len(b) > n+1 {
```

Comment:
So it looks like the `Reason-Phrase` from the HTTP RFC is entirely optional (https://stackoverflow.com/questions/17517086/can-an-http-response-omit-the-reason-phrase), 

So, if we force the h.SetStatusLine to always get called then either:

1) we have to throw an error if the phrase is omitted (which causes the `testResponseHeaderReadSuccess` in `header_test.go` to fail) or
2) fasthttp panics if we just always call `h.SetStatusLine(b[n+1:])` and the `Reason-Phrase` is omitted

Because of this, I don't think we can throw an error if `len(b) < n+1`, and we can't just assume that `SetStatusMessage` is always called when this function returns without an error.

To get around this, I've modified the `formatStatusLine` function so that if the given `statusText` is nil, it grabs a default status message for the given `statusCode` using the the `StatusMessage` function:

```go
func formatStatusLine(dst []byte, protocol []byte, statusCode int, statusText []byte) []byte {
	dst = append(dst, protocol...)
	dst = append(dst, ' ')
	dst = strconv.AppendInt(dst, int64(statusCode), 10)
	dst = append(dst, ' ')
	if statusText == nil {
		dst = append(dst, s2b(StatusMessage(statusCode))...)
	} else {
		dst = append(dst, statusText...)
	}
	return append(dst, strCRLF...)
}
```

Reply:
Sounds good, the only thing is that you need to check if `len(statusText) == 0` because `[]byte{} == nil` is `false` and the slice can be empty if the header is being reused.